### PR TITLE
Heroku buildpack

### PR DESCRIPTION
**Pull request by shirishgoyal**
One step compilation for heroku integrating node.js and django buildpacks
Add compile steps for DB migrations and Compressor

compilation steps are as below
a) npm install
b) bower install
c) pip install
d) collectstatic (edited)
e) migrate
f) compress

https://fast-ocean-2154.herokuapp.com/login

Seperated templates into two groups to make compressor work
a) django_templates
b) angular templates

### CODE REVIEW

**Comment by dmorina** [comment]
You may set SECURE flags to true when as it's supposed to run over https

Set debug to False also if the 404 and 500 templates exist 
The rest lgtm

**Comment by shirishgoyal** [comment]
Will take care of django settings for security in a new PR. Let's keep this one only focused on Heroku deployment and compressor

**Comment by dmorina** [comment]
lgtm

**Comment by shirishgoyal** [comment]
Added Django secure and logging settings for production

**Comment by mbernst** [comment]
@nistala I think we want your review on this one too

File: csp/settings.py [hunk 1]
```diff
@@ -210,3 +210,90 @@
...
+    SECURE_PROXY_SSL_HEADER = ('HTTP_X_FORWARDED_PROTO', 'https')
+    SECURE_HSTS_SECONDS = 31536000
+    SECURE_HSTS_INCLUDE_SUBDOMAINS = True
+    SECURE_FRAME_DENY = True
+    SECURE_CONTENT_TYPE_NOSNIFF = True
+    SECURE_BROWSER_XSS_FILTER = True
+    SESSION_COOKIE_SECURE = False
+    CSRF_COOKIE_SECURE = True
+    SESSION_COOKIE_HTTPONLY = True
+    SECURE_SSL_REDIRECT = True
```
Comment (dmorina) [inline csp/settings.py]: looks good ;)

**Comment by nistala** [comment]
testing

File: csp/settings.py [hunk 2]
```diff
@@ -210,3 +210,90 @@
...
+
+
+# MANAGER CONFIGURATION
+# ------------------------------------------------------------------------------
+# See: https://docs.djangoproject.com/en/dev/ref/settings/#admins
+ADMINS = (
+    ("Shirish Goyal", 'shirish.goyal@gmail.com'),       # add more team members
```
Comment (nistala) [inline csp/settings.py]: let's set this to the group or the general alias?

Comment (mbernst) [inline csp/settings.py]: We can use daemo@cs.stanford.edu, which autoforwards to Slack.

**Comment by nistala** [comment]
Need to add these commands to the readme:
```heroku buildpacks:add --index 1 https://github.com/heroku/heroku-buildpack-python.git```
(adds the buildpack at index 1)
```heroku buildpacks:add --index 1 https://github.com/heroku/heroku-buildpack-nodejs.git```
(adds the buildpack at index 1, moving the former to index 2)

**Comment by nistala** [comment]
let's also remove the .buildpacks file.

**Comment by nistala** [comment]
2 comments otherwise LGTM

**Comment by shirishgoyal** [comment]
All feedback incorporated
Email and documentation updated

**Pull request merged**